core/browser_events.ts

```diff
@@ -187,7 +187,7 @@ export function isTargetInput(e: Event): boolean {
  * @returns True if right-click.
  */
 export function isRightButton(e: MouseEvent): boolean {
-  if (e.ctrlKey && userAgent.MAC) {
+  if ((e.metaKey || e.ctrlKey) && userAgent.MAC) {
```

Comment:
Of note, I left `e.ctrlKey` here in case there's a legacy Mac case that works with it. The comment mentions "Mac OS X", which implies from my brief wiki check (https://en.wikipedia.org/wiki/MacOS) that it's for `Mac OS X 10.7` or before, which is 2011 and before.  I can't test this with Browser Stack :)

Reply:
This doesn't work unfortunately.

What this does is make the context menu open if you press command+click, but that's not the behavior we want. We want it to open on ctrl+click (which mirrors the behavior elsewhere in the os). 

I think the information in the original issue is needed here. The pointerevent while holding ctrl on firefox is never fired, so this method is never even hit because no event listener is triggered. We need to listen for the `contextmenu` event to be fired. That should happen in all browsers (even chrome) so if we listen for the `contextmenu` event we should be able to remove some of the code that checks for right-click events.

Btw, the distinction between macOS and OS X is unlikely to be relevant. It was called OS X for so long that it just still gets called that even though it's out of date.

Reply:
Does this work for you on other browsers as well? According to the `ctrlClick` documentation, this ctrl + click shouldn't work on any browser:
https://developer.mozilla.org/en-US/docs/Web/API/MouseEvent/ctrlKey

> On Macintosh keyboards, this key is labeled the control key. Also, note that on a Mac, a click combined with the control key is intercepted by the operating system and used to open a context menu, so ctrlKey is not detectable on click events.

Reply:
Yeah, it works fine. In Chrome if you ctrl click you get 3 events: `pointerdown`, `mousedown`, and `contextmenu`. In all 3 events, `ctrlKey` is true

In Firefox you only get `mousedown` and `contextmenu` events but `ctrlKey` is still true in both of them.

So MDN must be wrong. Maybe macOS changed this at some point.

Reply:
Good to know. Bummer! Testing in Browserstack with firefox on `macOS14` works for me since it translates `ctrl` to `command`.

If `contextmenu` works, though, which is also what the `ctrlKey` documentation implied above and you confirmed, perhaps I can attach a contextmenu event listener on Mac and proceed from there.

Reply:
Actually, MDN might be correct that `ctrlKey` isn't detectable on `click` events. But we're not listening to `click` events, so that is irrelevant to us :)